### DISCUSSION TITLE
Alternative to custom query transform now that its unsupported >9.0.0

### DISCUSSION THREAD

**Discussion by flynneva** [squidfunk/mkdocs-material#4887] [Q&A]
While trying to upgrade to >9.0.0 I noticed that [support for a custom query transform js method was removed](https://github.com/squidfunk/mkdocs-material/releases/tag/9.0.0)

What is the alternative if we would still like this feature?
Are queries still passed through the separator regex then? (from my testing...they are not any more)

Any help / tips / advice would be greatly appreciated!

**Comment by flynneva** [comment]
For example on the `mkdocs-material` site if you [search for `test::search`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/?q=test::search) I'd expect it to return results about the two tokens (`test` and `search`) assuming the regex separator was applied to the query.

Instead the results are null and nothing is returned. This is the same results as what I am seeing for our site as well.

**Reply by squidfunk** [reply]
Thanks for asking! This was actually an experiment to see whether this functionality is still needed. As search has been rebuilt, the way the custom query transform was defined would not work anymore. However, we can now fully alter search behavior, so supplying a custom function via a third-party script is definitely something that we can consider re-adding.

> For example on the mkdocs-material site if you [search for test::search](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/?q=test::search) I'd expect it to return results about the two tokens (test and search) assuming the regex separator was applied to the query.

The query is now [correctly lexed according to lunr.js](https://github.com/squidfunk/mkdocs-material/blob/6119df5d56fcdc90e25c5b1a47f7d2eba77f589e/src/assets/javascripts/integrations/search/query/_/index.ts#L63-L84) syntax. #4884 reported something similar, which is where I'll investigate whether we can treat non-field values as text. I can't make any promises though, but custom query transforms converting `:` to whitespace might solve the problem at hand.

**Reply by squidfunk** [reply]
As an addendum, a look in the console:

<img width="567" alt="Bildschirmfoto 2023-01-19 um 22 11 09" src="https://user-images.githubusercontent.com/932156/213560252-c5f510c6-687d-4678-af7a-ddccfd90d561.png">
